torch/csrc/jit/codegen/cuda/utils.cpp

```diff
@@ -310,12 +310,12 @@ KernelIndexMode collectIndexMode(const at::ArrayRef<at::IValue>& inputs) {
           // accumulate based on the sign of stride
           if (tensor_input.stride(dim_i) > 0) {
             // Acuumulate positive stride
-            tensor_most_positive_index +=
-                (tensor_input.size(dim_i) - 1) * tensor_input.stride(dim_i);
+            tensor_most_positive_index += (tensor_input.size(dim_i) - 1) *
+                tensor_input.stride(dim_i) * tensor_input.element_size();
           } else {
             // Acuumulate negative stride
-            tensor_most_negative_index +=
-                (tensor_input.size(dim_i) - 1) * tensor_input.stride(dim_i);
+            tensor_most_negative_index += (tensor_input.size(dim_i) - 1) *
+                tensor_input.stride(dim_i) * tensor_input.element_size();
```

Comment:
WARNING: index mode change. Some 32-bit indexable kernels now will need 64-bit indexing.

Reply:
Do we really need this change?